mathesar_ui/src/systems/table-view/table-inspector/table/links/utils.ts

```diff
@@ -0,0 +1,56 @@
+import type {
+  JoinableTable,
+  JoinableTablesResult,
+} from '@mathesar/api/tables/joinable_tables';
+import { MissingExhaustiveConditionError } from '@mathesar/utils/errors';
+
+export type TableLinkType = 'in_this_table' | 'from_other_tables';
+export type TableLink = {
+  table: { id: number; name: string };
+  column: { id: number; name: string };
+};
+
+function generateTableLinkFromJoinableTable(
+  linkedTable: JoinableTable,
+  joinableTablesResult: JoinableTablesResult,
+): TableLink {
+  return {
+    table: {
+      id: linkedTable.target,
+      name: joinableTablesResult.tables[linkedTable.target].name,
+    },
+    column: {
+      id: linkedTable.jp_path[0].slice(-1)[0],
+      name: joinableTablesResult.columns[linkedTable.jp_path[0].slice(-1)[0]]
+        .name,
+    },
```

Comment:
Identification logic for column is incorrect. This logic works only for links from `from_other_tables`.

For links in the base table i.e `in_this_table`, the first element in the array represents the column id, not the second. This check is direction and `fk_path[0][2]` needs to be taken into account to determine which column id to pick.

Reply:
I see that @seancolsen has opened an issue which is caused by this incorrect logic: https://github.com/centerofci/mathesar/issues/1960.

I'd like the fix to be made on this PR itself. @rajatvijay Please mark the issue as fixed/closed once you've implemented the fix here.